be/src/fs/fs_s3.cpp

```diff
@@ -304,6 +304,12 @@ static std::shared_ptr<Aws::S3::S3Client> new_s3client(const S3URI& uri, const F
         }
         config.maxConnections = config::object_storage_max_connection;
     }
+    if (config::object_storage_connect_timeout_ms >= 0) {
```

Comment:
notice line:272, will not go here?

Reply:
> notice line:272, will not go here?

Yes, I noticed this. I think we only need to modify the configuration items when creating a new one. For a whole configuration object passed by users, it's better to keep the original attributes, rather than overwrite some items which may cause some unexpected problems.

Reply:
If the user is using `aws.s3.access_key` and `aws.s3.secret_key`, that means the code will not go here.

Reply:
> If the user is using `aws.s3.access_key` and `aws.s3.secret_key`, that means the code will not go here.

OK